### PR TITLE
[#72] Add secure service for TLS support

### LINKED ISSUE #72: Create a secured service to expose the application service over TLS inside the cluster
**Is your feature request related to a problem? Please describe.**

The Helm Chart create a service to expose the application inside the cluster. This service uses the HTTP port of the server.
It should be possible to create a secured service that use the HTTPS port of the server (8443) so that the connection would use TLS through the whole path from the user to the container (inside and outside the cluster).

Add a `deploy.tls.enabled` boolean field that would trigger the creation of a service exposing the HTTPS port and use it to route the traffic from the route (if `deploy.route.tls.enabled` is also `true`).
The regular HTTP service would still be created regardless of the value of `deploy.tls.enabled`.

### PR DESCRIPTION
**Pull request by jmesnil**
* Add deploy.tls.enabled to enable the access of WildFly over HTTPS
* Add the `https` example to demonstrate how to configure WildFly to use
  HTTPS

This fixes #72.

Signed-off-by: Jeff Mesnil <jmesnil@redhat.com>

### CODE REVIEW

**Comment by jmesnil** [comment]
@yersan Could you please review?
@jfdenise Could you please review? This use case is the first time we require to use the cloud galleon pack to get access to the S2I env vars to configure HTTPS.

File: examples/https/README.adoc [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+Let's specify these HTTPS environment variables when the application image is deployed:
+
+[source,yaml]
+----
+deploy:
+  env:
+    - name: HTTPS_KEYSTORE
+      value: keystore.jks
+    - name: HTTPS_KEYSTORE_TYPE
```
Comment (jfdenise) [inline examples/https/README.adoc:100]: This env is not rquired, elytron will discover that the keystore is JKS

File: examples/https/README.adoc [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+          key: keystore-password
+          name: wildfly-app-secret
+    - name: HTTPS_KEYSTORE_DIR
+      value: /etc/wildfly-app-secret
+----
+
+These 4 environment variables must be specified:
+
+* `HTTPS_KEYSTORE` is the name of the keystore file (`keystore.jks``)
+* `HTTPS_KEYSTORE_TYPE` is the type of keystore (`JKS`)
```
Comment (jfdenise) [inline examples/https/README.adoc:114]: This env is not rquired, elytron will discover that the keystore is JKS

File: examples/https/helm.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+build:
+  uri: https://github.com/wildfly/quickstart.git
+  ref: 26.1.0.Final
+  contextDir: helloworld-rs
+  s2i:
+    featurePacks:
+      - org.wildfly:wildfly-galleon-pack:26.1.1.Final
+      - org.wildfly.cloud:wildfly-cloud-galleon-pack:1.0.1.Final
```
Comment (jfdenise) [inline examples/https/helm.yaml:8]: Use 1.1.1.Final

File: examples/https/helm.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  tls:
+    enabled: true
+  route:
+    tls:
+      enabled: true
+      termination: passthrough
+  env:
+    - name: HTTPS_KEYSTORE
+      value: keystore.jks
+    - name: HTTPS_KEYSTORE_TYPE
```
Comment (jfdenise) [inline examples/https/helm.yaml:21]: This env is not rwquired, elytron will discover that the keystore is JKS

**Review by jfdenise — CHANGES_REQUESTED** [review]

File: examples/https/README.adoc (hunk 1 above)
Comment (jmesnil) [reply examples/https/README.adoc:100]: thanks, it works without it!

@jfdenise However this is not how I understood its docs at https://github.com/wildfly/wildfly-cekit-modules/blob/5a2927a952d07fd996fabd7f99d1b56d20053947/jboss/container/wildfly/launch/elytron/module.yaml#L31.

> The underlying type of the keystore (JKS, JCEKS, etc.). Defaults to JCEKS

The doc should be updated to specify that if not defined, Elytron will automatically discover the the type of the keystore.

File: examples/https/README.adoc (hunk 2 above)
Comment (jmesnil) [reply examples/https/README.adoc]: fixed, thanks

File: examples/https/helm.yaml (hunk 3 above)
Comment (jmesnil) [reply examples/https/helm.yaml]: fixed

**Review by jfdenise — APPROVED** [review]

**Pull request merged**